fastapi/applications.py

```diff
@@ -93,6 +94,11 @@ def __init__(
             [] if middleware is None else list(middleware)
         )
         self.middleware_stack: ASGIApp = self.build_middleware_stack()
+        
+        def clear_lifetime_dependencies():
+            self.lifetime_dependencies = {}
+
+        self.on_event("shutdown")(clear_lifetime_dependencies)
```

Comment:
Clearing the dependencies at shutdown aught to help with garbage collection by releasing the refs.